### PR TITLE
Api V2

### PR DESCRIPTION
**Pull request by jtnegrotto**
This is a complete rewrite for v2. I've differed in a few ways from our SDK guidelines for the sake of an idiomatic API, e.g. having only one params hash for arguments so that `Fax.create to: '...', api_key: '1234'` would work without needing additional commas and braces.

Looks like the build is failing for 1.9.3 because the version of bundler they're running thinks dependencies on a particular version of Ruby are for a gem called `ruby`. ~Also, tests aren't being Run. Need to add a Rake task for running them.~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,101 +1,288 @@
...
+Fax.test_receive file: fax_file
+# => true
+```
+
+##### `Fax.supported_countries`
+
+Get a list of supported countries.
+
+``` ruby
+Fax.supported_countries
```
Comment (jpettettphaxio) [inline README.md]: In the other v2 client libs, we do something like `Public::Country.list` to more closely match the API path.

File: README.md [hunk 2]
```diff
@@ -1,101 +1,288 @@
...
+##### `PhoneNumber.delete`
+
+Release a phone number.
+
+``` ruby
+PhoneNumber.delete '+15558675309'
+# => true
+```
+
+##### `PhoneNumber.list_available_area_codes`
```
Comment (jpettettphaxio) [inline README.md]: How about `Public::AreaCode.list`?

**Comment by jtnegrotto** [comment]
@jpettettphaxio Those changes have been made.

File: README.md [hunk 3]
```diff
@@ -1,101 +1,289 @@
...
+
+Test receiving a fax.
+
+``` ruby
+fax_file = File.open 'test.pdf', 'rb'
+Fax.test_receive file: fax_file
+# => true
+```
+
+#### `Public::Country`
```
Comment (jpettettphaxio) [inline README.md]: I think this should be:
```
#### Countries

##### `Public::Country.list`
```

File: README.md [hunk 4]
```diff
@@ -1,101 +1,289 @@
...
+Release a phone number.
+
+``` ruby
+PhoneNumber.delete '+15558675309'
+# => true
+```
+
+#### Area Codes
+
+##### `Public::AreaCode`
```
Comment (jpettettphaxio) [inline README.md]: Should be `Public::AreaCode.list`?

File: README.md [hunk 5]
```diff
@@ -1,101 +1,289 @@
...
+
+Lists available area codes for purchasing Phaxio numbers.
+
+``` ruby
+area_codes = Public::AreaCode.list toll_free: true
+# => Phaxio::Resource::Collection([Public::AreaCode(city: 'Chicago', ...), ...], page: 1)
```
Comment (jpettettphaxio) [inline README.md]: This is syntactically correct, but when toll_free is true, city will always be "Toll Free Service"

File: spec/support/vcr_cassettes/resources/public/area_code/list.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
+---
```
Comment (jpettettphaxio) [inline spec/support/vcr_cassettes/resources/public/area_code/list.yml]: This file appears to be an unused duplicate of area_codes/list.yml

**Pull request merged**